Eshop.Domain/Product.cs

```diff
@@ -2,26 +2,29 @@
 {
     public class Product
     {
+        private Product() { } // This constuctor is only for EF
+
         public Product(int id, string title, string description, decimal price, Category category)
         {
             if (id < 0)
             {
-                throw new ArgumentOutOfRangeException("id");
+                throw new ArgumentOutOfRangeException(nameof(id));
             }
 
             ValidateParameters(title, description, price, category);
 
-            Id = id;
+            Id = id; 
             Title = title;
             Description = description;
             Price = price;
             Category = category;
         }
 
-        public int Id { get; }
+        public int Id { get; }  
         public string Title { get; private set; }
         public string Description { get; private set; }
         public decimal Price { get; private set; }
+        public int CategoryId { get; set; }
```

Comment:
public set again
1 think is navigation property and second you can not ruin Encapsulation.